source/big_core/illegal_instructions.vh

```diff
@@ -50,7 +43,7 @@ assign illegal_instructionsQ101H.RopFunct7NotMatchZero = (PreOpcodeQ101H == R_OP
                                                                                                                          || PreRtypeQ101H  == AND_) && (PreFunct7Q101H != '0)  : 1'b0;      
                                                                                                                                                         
 // Covers R_type instructions where bits [31:25] must always be 7'b0100000 = 0x20 or zero in Base ISA //TODO - some extentions may have differente options 
-assign illegal_instructionsQ101H.RopFunct7NotMatch20OrZero = (PreOpcodeQ101H == R_OP && MultiplyInstructionsQ101H) ? 1'b0 : (PreOpcodeQ101H == R_OP) ?
+assign illegal_instructionsQ101H.RopFunct7NotMatch20OrZero = (PreOpcodeQ101H == R_OP && PreFunct7Q101H == 32'h1) ? 1'b0 : (PreOpcodeQ101H == R_OP) ?
```

Comment:
I think we should have a simpler "illegal_instruction" detection.. something more robust and scalable..

I might have an idea - its will using the enum that has a value for each valid instruction. and if the value does not match any valid instruction - its an illegal instruction